src/presenced.cpp

```diff
@@ -342,21 +355,36 @@ Client::reconnect(const std::string& url)
                                                     (struct sockaddr *)&ipv4addr, sizeof(ipv4addr),
                                                     mUrl.port, (mUrl.path).c_str())), "connect");
                 }
+                return promise::_Void();
             })
-            .fail([this](const promise::Error& err)
+            .fail([wptr, this](const promise::Error& err)
             {
-                int errType =  (err.type() == ERRTYPE_MEGASDK) ? WS_ERRTYPE_DNS : err.type();
-                mConnectPromise.reject(err.msg(), err.code(), errType);
-                mLoginPromise.reject(err.msg(), err.code(), errType);
+                PRESENCED_LOG_ERROR("Connect error: %s", err.what());
+                if (wptr.deleted())
+                    return;
+                auto errtype = (err.type() == ERRTYPE_MEGASDK) ? WS_ERRTYPE_DNS : err.type();
+                if (!mConnectPromise.done())
+                    mConnectPromise.reject(err.msg(), err.code(), errtype);
+                if (!mLoginPromise.done())
+                    mLoginPromise.reject(err.msg(), err.code(), errtype);
             });
```

Comment:
Yes, gcc. The warning says something like _non-void function should return a value_